tests/testthat/test-translate-sql-string.R

```diff
@@ -24,6 +24,9 @@ test_that("sql_str_sub works as expected", {
   expect_equal(substr(x, -1), sql("SUBSTR(`x`, -1)"))
   expect_equal(substr(x, 2, 4), sql("SUBSTR(`x`, 2, 3)"))
   expect_equal(substr(x, 2, 2), sql("SUBSTR(`x`, 2, 1)"))
+  expect_equal(substr(x, 2, 0), sql("SUBSTR(`x`, 2, 0)"))
```

Comment:
Given the challenge of getting this right, would you mind adding a couple of tests that use `memdb_frame()` to actually run the code and check that the result is equivalent to R?